pkg/openapi/manager.go

```diff
@@ -166,7 +153,7 @@ func (o *manager) ValidatePolicyMutation(policy kyvernov1.PolicyInterface) error
 		}
 
 		if kind != "*" {
-			err = o.ValidateResource(*patchedResource.DeepCopy(), "", kind)
+			err = o.validateResource(*patchedResource.DeepCopy(), "", kind)
```

Comment:
LGTM! But I don't understand why we are using locks here

Reply:
Because a lock can not be acquired multiple times by the same go routine. So we acquire locks in public methods (or let the caller manage the lock) and make private methods lock free.

The rule of thumb is that public methods don't call other public methods, only private ones to prevent deadlocks...

Reply:
Got it, thanks @eddycharly!